### PR TITLE
Refactors radiation collectors, allow them to make cargo money

### PR DESCRIPTION
**Pull request by redmoogle**
# General Documentation

### Intent of your Pull Request

Refactors the shit out of them to make adding modes easier.
Also adds the ability to make money with them(150min per collector on normal SM)

### Why is this change good for the game?

Easier to work with code, also feature

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
Radiation collector can make money now

### What general grouping does this PR fall under? 
Engineering changes

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
Money mode makes about ~150 credits per minute per collector

# Changelog

:cl:  
rscadd: Radiation collectors now have a credit mode
/:cl:

### CODE REVIEW

**Comment by ChesterTheCheesy** [comment]
Are tesla coils gonna get the same treatment?

**Comment by redmoogle** [comment]
> Are tesla coils gonna get the same treatment?

Maybe

**Comment by redmoogle** [comment]
Still working on making the money part less fucky and not giving billions of credits for a rad burn

**Comment by redmoogle** [comment]
since this realistically will be reviewed in 10 years im just closing it
